src/tapeimage.cpp

```diff
@@ -215,6 +212,12 @@ void tapeimage::read_header() noexcept (false) {
                 "recovery not implemented"
             );
 
+        case LFP_EOF:
+        {
+            const auto msg = "tapeimage: unexpected EOF when reading header "
+                                "- got {} bytes";
+            throw protocol_fatal(fmt::format(msg, n));
```

Comment:
(comments follows the tests I've been writing, maybe doesn't even have to be addressed in this PR, but leaving here as related to change)

Throwing fatal error in all these cases might not be the best idea.
Imagine we have a truncated file (so no nice EOF tapemarks in the end).
- If file is truncated inside of the actual data, we will nicely return LFP_EOF and whatever we were able to read. User won't even know that TIF protocol was so severely truncated.
- But if file happened to be truncated inside header, we suddenly scream protocol fatal
- (actual 3 cases) And if the data in file is actually nice and complete, just final file headers are missing (we don't know that and try to read new header) - we again suddenly scream fatal.

Model seems somewhat inconsistent to me. But again, probably this doesn't have to do much with this commit.